helper/users/cache.go

```diff
@@ -0,0 +1,81 @@
+package users
+
+import (
+	"os/user"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/nomad/lib/lang"
+	"oss.indeed.com/go/libtime"
+)
+
+const (
+	cacheTTL   = 1 * time.Hour
+	failureTTL = 1 * time.Minute
+)
+
+type entry[T any] lang.Pair[T, time.Time]
+
+func (e *entry[T]) expired(now time.Time, ttl time.Duration) bool {
+	return now.After(e.Second.Add(ttl))
+}
+
+type (
+	userCache        map[string]*entry[*user.User]
+	userFailureCache map[string]*entry[error]
```

Comment:
Any concerns about unlimited memory growth? I don't expect these to grow too much, and creating another GC routine is a bit much 😅  but maybe we can delete entries on read once they expire?

Reply:
Oh yeah that's a nice idea. In practice this _should_ always be zero, but yeah people certainly do stuff to their clusters.